apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextManager.java

```diff
@@ -224,4 +226,17 @@ public static RuntimeContext getRuntimeContext() {
 
         return runtimeContext;
     }
+
+    /**
+     * Get current context profiling status
+     * @return
+     */
+    public static ProfilingStatus getProfilingStatus() {
+        final AbstractTracerContext context = get();
```

Comment:
This will cost unexpected resource. We should use `span` to delegate. Such as `span#isProfiling`.